openedx/features/edly/tests/test_middleware.py

```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+
+        self.client = Client(SERVER_NAME=self.request.site.domain)
+        self.client.cookies.load(
+            {
+                settings.EDLY_USER_INFO_COOKIE_NAME: cookies._get_edly_user_info_cookie_string(self.request)
+            }
+        )
+        self.client.login(username=self.user.username, password='test')
+
+    def test_settings_override_middleware_logs_warning_if_no_site_configuration_is_present(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if no site configuration is present.
+        """
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site ({site}) has no related site configuration.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_values_set(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if site configuration has no values.
+        """
+        SiteConfigurationFactory(site=self.request.site)
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site configuration for site ({site}) has no values set.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_django_settings_overrides(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if site configuration has no django settings override values.
+        """
+        SiteConfigurationFactory(
+            site=self.request.site,
+            values={
+                'PLATFORM_NAME': 'St. NormanX',
+            }
+        )
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site configuration for site ({site}) has no django settings overrides.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_overrides_settings_correctly(self):
+        """
+        Tests "SettingsOverrideMiddleware" correctly overrides django settings.
+        """
+        SiteConfigurationFactory(
+            site=self.request.site,
+            values={
+                'DJANGO_SETTINGS_OVERRIDE': {
+                    'EDLY_USER_INFO_COOKIE_NAME': 'edly-user-info-test'
+                }
+            }
+        )
+        assert settings.EDLY_USER_INFO_COOKIE_NAME == 'edly-user-info'
+        self.client.get('/', follow=True)
+        assert settings.EDLY_USER_INFO_COOKIE_NAME == 'edly-user-info-test'
```

Comment:
Since we will need to validate multiple values so create a helper method (https://github.com/edly-io/edly-panel-backend/pull/6/files):
```python
def _assert_response_settings_values(self, response, expected_site_configuration_values):
    """
    Validate configuration settings in HTTP response.
    """
    for config_key, expected_config_value in expected_site_configuration_values.items():
        assert expected_config_value == getattr(response.context['settings'], config_key)
```

Reply:
done